tests/rules/avoid-cyclic-effects.test.ts

```diff
@@ -234,19 +296,57 @@ ruleTester().run(path.parse(__filename).name, rule, {
             this.actions$.pipe(
             ~~~~~~~~~~~~~~~~~~ [${messageId}]
               ofType(genericFoo),
+              mergeMapTo(of(genericFoo()))
```

Comment:
Could give me a hand here please? I have no idea why this test if failing (and yes, I checked the imports 😅).

Note that `mergeMapTo([...])` and `mergeMap(() => somethingElse)` are reported as expected, but this one here isn't 🤔

Reply:
It has something to do with the effect config parameter, if I set it to `{dispatch:true}` the test succeeds.
If I change another test, and remove that parameter that test also fails, e.g.

```diff
@Injectable()
      class Effect {
        foo$: CreateEffectMetadata
        constructor(customActions$: Actions) {
          this.foo$ = createEffect(() =>
            customActions$.pipe(
            ~~~~~~~~~~~~~~~~~~~ [${messageId}]
              ofType(genericFoo),
              switchMapTo([genericBar(), genericFoo()]),
            ),
-            { useEffectsErrorHandler: true }
+         configDispatchTrue // or don't provide a config and see it fail too
          )
        }
      }
```

Reply:
Heh... this is strange.
I have to go now but I'll take another look later/tomorrow.

Reply:
Yep, it's something iffy...
I didn't had the time to take a close look, I'll try to do it this week though.